05-http-server-streams/01-file-server-get/server.js

```diff
@@ -20,4 +21,35 @@ server.on('request', (req, res) => {
   }
 });
 
+function sendFile(res, pathname, filepath) {
+  if (pathname.includes('/')) {
+    res.statusCode = 400;
+    res.end('Subfolders are not supported');
+
+    return;
+  }
+
+  fs.stat(filepath, (error, stats) => {
+    if (error || !stats.isFile()) {
+      res.statusCode = 404;
+      res.end('File not found');
+
+      return;
+    }
+
+    const file = new fs.ReadStream(filepath);
+
+    file.on('error', () => {
+      res.statusCode = 500;
+      res.end('Something went wrong');
+    });
+
+    file.pipe(res);
+
+    res.on('close', () => {
```

Comment:
При успешном окончании pipe нет необходимости уничтожать поток 
https://nodejs.org/api/stream.html#stream_readable_pipe_destination_options

> By default, stream.end() is called on the destination Writable stream when the source Readable stream emits 'end', so that the destination is no longer writable.